yb-voyager/cmd/exportData_test.go

```diff
@@ -0,0 +1,660 @@
+		msr.TableListExportedFromSource = lo.Map(expectedTableListWithOnlyRootTable, func(t sqlname.NameTuple, _ int) string {
+			return t.ForOutput()
+		})
+		msr.SourceExportedTableListWithLeafPartitions = lo.Map(expectedTableList, func(t sqlname.NameTuple, _ int) string {
+			return t.ForOutput()
+		})
+		msr.SourceRenameTablesMap = expectedPartitionsToRootMap
+	})
+	if err != nil {
+		t.Fatalf("error updating msr: %v", err)
+	}
+
+	testCasesWithDifferentTableListFlagValuesTest1(t, expectedTableList, expectedPartitionsToRootMap)
+
+}
+
+func testCasesWithDifferentTableListFlagValuesTest1(t *testing.T, firstRunTableList []sqlname.NameTuple, firstRunPartitionsToRootMap map[string]string) {
+	//case1: getInitialTableList for subsequent run with  start-clean false and basic with same table-list flags so no guardrails
+	startClean = false
+
+	getInitialTableistAndAssertExpectedResult(t, firstRunTableList, firstRunPartitionsToRootMap)
+
+	//case2: getInitialTableList for subsequent run with  start-clean false and basic with no table-list flags so no guardrails
+	startClean = false
+	source.TableList = ""
+	source.ExcludeTableList = ""
+
+	getInitialTableistAndAssertExpectedResult(t, firstRunTableList, firstRunPartitionsToRootMap)
+
+	//case3: getInitialTableList for subsequent run with  start-clean false and basic with table-list flag
+	startClean = false
+	//--table-list test_partitions_sequences,sales_region
+	source.TableList = "test_partitions_sequences,sales_region"
+	utils.DoNotPrompt = true
+
+	rootTables := []sqlname.NameTuple{
+		getNameTuple("public.test_partitions_sequences"),
+		getNameTuple("public.sales_region"),
+	}
+
+	expectedMissingTables := []sqlname.NameTuple{
+		getNameTuple("public.datatypes1"),
+		getNameTuple("public.foreign_test"),
+	}
+
+	expectedExtraTables := []sqlname.NameTuple{
+		getNameTuple("p1.london"),
+		getNameTuple("p1.sydney"),
+	}
+
+	assertGuardrailsChecksForMissingAndExtraTablesInSubsequentRun(t, expectedMissingTables, expectedExtraTables, firstRunTableList, rootTables)
+	getInitialTableistAndAssertExpectedResult(t, firstRunTableList, firstRunPartitionsToRootMap)
+
+	//case4: getInitialTableList for subsequent run with  start-clean false and basic with only exclude-table-list flag
+	startClean = false
+	//--exclude-table-list sales_region,foreign_test
+	source.TableList = ""
+	source.ExcludeTableList = "sales_region,foreign_test"
+	utils.DoNotPrompt = true
+
+	expectedMissingTables2 := []sqlname.NameTuple{
+		getNameTuple("p1.boston"),
+		getNameTuple("public.foreign_test"),
+	}
+
+	expectedExtraTables2 := []sqlname.NameTuple{}
+
+	assertGuardrailsChecksForMissingAndExtraTablesInSubsequentRun(t, expectedMissingTables2, expectedExtraTables2, firstRunTableList, rootTables)
+	getInitialTableistAndAssertExpectedResult(t, firstRunTableList, firstRunPartitionsToRootMap)
+
+	//case5: getInitialTableList for subsequent run with  start-clean false and basic with table-list and exclude-table-list flags
```

Comment:
is passing both include and exclude list allowed today? i would have thought we should disallow it? in what situations does this help?

Reply:
yes we allow, it helps in scenario where:
tables on source - ab1,abc2,abc3,abc4,abc5, <few other tables>
table-list abc* (meaning all the abc prefixed tables)
exclude-table-list abc5 (excluding the abc5)